docs/tutorials/talk-to-kafka.rst

```diff
@@ -0,0 +1,715 @@
+That's because the consumer instance is listening for messages and currently, there's no message to print out. 
+To see some action on this terminal, run the producer code in another terminal. You will see the same record appear on the producer program terminal.
+
+What would happen if there were two partitions in this case, ``p0`` and ``p1``? In this case, messages would be published to partition randomly. The consumer ``co0`` would take a round robin approach when consuming messages from this topic.
+
+1 topic : 1 partition : 2 consumers : 1 consumer group
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's take a look at this setup where there are two consumers. ``co0`` and ``co1`` are registered to the same `consumer group A`. 
+
+In this setup, one consumer will be sitting idle. This highlights an important concept in Kafka - records are processed in parallel and same partition cannot be assigned to multiple consumers in the same consumer group.
+
+.. mermaid::
+    
+    graph LR;
+
+        pr0(kafka producer pr0) -->p0(partition p0);
+        subgraph topic
+        p0
+        end
+        co0(kafka consumer co0)
+        co1(kafka consumer co1)
+        subgraph consumer group A
+        co0
+        co1
+        end
+        p0 -->co0 
+
+If the first consumer ``co0`` crashes for some reason, the other consumer ``co1`` in the consumer group will begin consuming messages from the last committed offset of the partition. 
+
+.. mermaid::
+    
+    graph LR;
+
+        pr0(kafka producer pr0) -->p0(partition p0);
+        subgraph topic
+        p0
+        end
+        co0(CRASHED)
+        co1(kafka consumer co1)
+        subgraph consumer group A
+        co0
+        co1
+        end
+        p0 -->co1
+
+Add schemas to messages with Apache Avro™
+-------------------------------------------
+
+The Kafka brokers understand data as stream of bytes so one needs to pick a serializer and deserializer to convert the bytes into meaningful messages. 
+Any format would do as long as it's consistent. For this tutorial, we're selecting Apache Avro which is an open-source project and one of the most popular serialization format.
+Avro is defined by a schema and the schema is written in JSON. You can consider Avro as JSON with a schema attached to it.
+
+Both of these programs use SSL authentication. You can follow the previous examples of this tutorial if you'd like to use SASL authentication instead.
+
+Setting up a producer to send Avro-encoded messages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here's an example of a Kafka consumer in Python using the `confluent-Kafka-python <https://github.com/confluentinc/confluent-kafka-python>`_ library to produce Avro-encoded messages.
+On a terminal window, run the following producer code. The program should execute and exit. You won't see anything yet since we need the consumer program to run next. 
```

Comment:
```suggestion
In a terminal window, run the following producer code. The program should execute and exit. You won't see anything yet since we need the consumer program to run next. 
```